Floofbot/Modules/Administration.cs

```diff
@@ -133,43 +127,22 @@ public async Task warnlog(string user)
                 return;
             }
 
-            string sql = @"SELECT DateAdded,Moderator,Reason FROM Warnings
-                 WHERE UserID = $UserId AND $GuildId = $GuildId ORDER BY Id desc";
-            SqliteCommand command = new SqliteCommand(sql, dbConnection);
-            command.Parameters.Add(new SqliteParameter("$UserId", badUser.Id.ToString()));
-            command.Parameters.Add(new SqliteParameter("$GuildId", Context.Guild.Id.ToString()));
-
-            dbConnection.Open();
-            var results = command.ExecuteReader();
+            var warnings = _floofDB.Warnings.AsQueryable()
+                .Where(u => u.UserId == badUser.Id && u.GuildId == Context.Guild.Id)
+                .OrderByDescending(x => x.Id);
 
             EmbedBuilder builder = new EmbedBuilder();
-            if (results.HasRows) {
-                builder.Title = $"Warnings for {badUser.Username}#{badUser.Discriminator}";
-                builder.Color = Color.DarkOrange;
-
-                //discord embeds have a limit of 25 fields.  Shows the most recent 25.
-                int warningCount = 0;
-
-                while (results.Read()) {
-                    var warning = new {
-                        DateAdded = results.GetValue(results.GetOrdinal("DateAdded")).ToString(),
-                        Moderator = results.GetValue(results.GetOrdinal("Moderator")).ToString(),
-                        Reason = results.GetValue(results.GetOrdinal("Reason")).ToString(),
-                    };
-
-                    builder.AddField($"**{warningCount + 1}**.  {DateTime.Parse(warning.DateAdded).ToString("MMMM dd yyyy")} by {warning.Moderator}", $"```{warning.Reason}```");
-                    warningCount++;
-
-                    //if we reach more then 25 warnings for a user then we are doing something wrong
-                    if (warningCount > 24)
-                        break;
-                }
-                await Context.Channel.SendMessageAsync("", false, builder.Build());
+            int warningCount = 0;
+            builder.WithTitle($"Warnings for {badUser.Username}#{badUser.Discriminator}");
+            foreach (Warning warning in warnings) {
+                builder.AddField($"**{warningCount + 1}**. {warning.DateAdded.ToString("MMMM dd yyyy")}", $"```{warning.Reason}```");
+                warningCount++;
+
+                //discord embeds have a limit of 25 fields
+                if(warningCount > 24) 
+                    break;
```

Comment:
You could also just limit the number of rows returned by the DB query
or you could make it just generate a new embed with the left-over warnings